tests/cmake/tpls.cmake

```diff
@@ -0,0 +1,170 @@
+
+    # # when trilinos is also enabled it links kokkos too, see tplTrilinos.cmake
+    if(NOT PRESSIO_ENABLE_TPL_TRILINOS)
+      # if kokkos is used as standalone lib, then we are more specific
+      # user needs to defined: KOKKOS_ROOT_DIR and KOKKOS_KERNELS_ROOT_DIR
+      if (NOT KOKKOS_ROOT OR NOT KOKKOS_KERNELS_ROOT)
+	message(FATAL_ERROR "Missing KOKKOS_ROOT. KOKKOS needs:
+          -D KOKKOS_ROOT=<full-path-to-kokkos-installation>
+          -D KOKKOS_KERNELS_ROOT=<full-path-to-kokkos-kernels-installation>
+          ")
+      endif()
+
+      set(KOKKOS_LIB_NAMES kokkoscontainers kokkoscore kokkoskernels)
+
+      include_directories(${KOKKOS_ROOT}/include ${KOKKOS_KERNELS_ROOT}/include)
+
+      link_directories(${KOKKOS_ROOT}/lib ${KOKKOS_ROOT}/lib64
+	${KOKKOS_KERNELS_ROOT}/lib ${KOKKOS_KERNELS_ROOT}/lib64)
+
+      link_libraries(${KOKKOS_LIB_NAMES})
+    endif()
+
+  endif()
+endif()
+
+
+if(PRESSIO_ENABLE_TPL_TRILINOS)
+  if(PRESSIO_ENABLE_UNIT_TESTS OR PRESSIO_ENABLE_TESTS)
+    if (NOT TRILINOS_ROOT)
+      message(FATAL_ERROR
+	"You enabled PRESSIO_ENABLE_TPL_TRILINOS but did not set TRILINOS_ROOT.
+        Please reconfigure with:
+          -D TRILINOS_ROOT=<full-path-to-trilinos-install>
+          ")
+    endif()
+
+    set(TRILINOS_LIB_NAMES kokkosalgorithms
+      kokkoscontainers
+      kokkoscore
+      teuchoskokkoscomm
+      teuchoskokkoscompat
+      teuchosremainder
+      teuchosnumerics
+      teuchoscomm
+      teuchosparameterlist
+      teuchosparser
+      teuchoscore
+      epetra
+      epetraext
+      ifpack
+      aztecoo
+      tpetraext
+      tpetrainout
+      tpetra
+      kokkostsqr
+      tpetraclassiclinalg
+      tpetraclassicnodeapi
+      tpetraclassic
+      kokkoskernels
+      ifpack2
+      triutils
+      # repeat to solve issue we have on linux
+      kokkosalgorithms
+      teuchosparameterlist)
+
+    include_directories(${TRILINOS_ROOT}/include)
+    link_directories(${TRILINOS_ROOT}/lib ${TRILINOS_ROOT}/lib64)
+    link_libraries(${TRILINOS_LIB_NAMES})
+  endif()
+endif()
+
+
+# function(pybind11_fatal)
+#   message(FATAL_ERROR "I cannot find the Pybind11 library.
+# To use it, please reconfigure with:
+# -D PYBIND11_ROOT=<full-path-to-installation>")
+# endfunction()
+
+# if(PRESSIO_ENABLE_TPL_PYBIND11)
+#   message("Found PRESSIO_ENABLE_TPL_PYBIND11=${PRESSIO_ENABLE_TPL_PYBIND11}.")
+
+#   # if we need to build tests, then prep for it
+#   if(PRESSIO_ENABLE_UNIT_TESTS OR PRESSIO_ENABLE_TESTS)
+
+#     # if PYBIND11_ROOT not found
+#     if (NOT PYBIND11_ROOT)
+#       pybind11_fatal()
+#     endif()
+
+#     find_package(pybind11 REQUIRED PATHS ${PYBIND11_ROOT}/share/cmake)
+#     find_package(Python3 COMPONENTS Interpreter NumPy)
+
+#     include_directories(${Python3_INCLUDE_DIRS} ${PYBIND11_ROOT}/include)
+
+#     if(NOT ${Python3_FOUND})
+#       message(FATAL_ERROR "Python > 3 not found")
+#     endif()
+#   endif()
+# endif()
```

Comment:
let's not worry about pybind11 for now, you can skip all of it for now

Reply:
Skipped that without removing the comments. We might purge them in subsequent PR to make following the history easier.